src/status_im/contexts/wallet/wallet_connect/modals/base_modal/view.cljs

```diff
@@ -0,0 +1,114 @@
+(ns status-im.contexts.wallet.wallet-connect.modals.base-modal.view
+  (:require [quo.core :as quo]
+            [quo.foundations.colors :as colors]
+            [react-native.core :as rn]
+            [react-native.safe-area :as safe-area]
+            [status-im.common.raw-data-block.view :as raw-data-block]
+            [status-im.common.standard-authentication.core :as standard-authentication]
+            [status-im.contexts.wallet.wallet-connect.modals.base-modal.style :as style]
+            [utils.i18n :as i18n]
+            [utils.re-frame :as rf]))
+
+(defn- close-sheet
+  []
+  (rf/dispatch [:navigate-back]))
+
+(defn- on-auth-success
+  [password]
+  (rf/dispatch [:hide-bottom-sheet])
+  (rf/dispatch [:wallet-connect/respond-current-session password]))
+
+(defn data-block
+  []
+  (let [display-data (rf/sub [:wallet-connect/current-request-display-data])]
+    [raw-data-block/view
+     {:data           display-data
+      :bottom-margin? false}]))
+
+(defn header
+  [{:keys [label dapp account]}]
+  [rn/view
+   {:style style/header-container}
+   [quo/text
+    {:size   :heading-1
+     :weight :semi-bold}
+    (let [{:keys [name icons]} (:peerMetadata dapp)]
+      [rn/view {:style style/header-dapp-name}
+       [quo/summary-tag
+        {:type         :dapp
+         :label        name
+         :image-source (first icons)}]])
+    (str " " label " ")
+    (let [{:keys [emoji customization-color name]} account]
+      [rn/view {:style {:padding-top 4}}
+       [quo/summary-tag
+        {:type                :account
+         :emoji               emoji
+         :label               name
+         :customization-color customization-color}]])]])
+
+(defn view
+  [{:keys [header-label warning-label show-network? slide-button-text]}]
+  (let [bottom                (safe-area/get-bottom)
+        {:keys [customization-color]
+         :as   account}       (rf/sub [:wallet-connect/current-request-account-details])
+        dapp                  (rf/sub [:wallet-connect/current-request-dapp])
+        network               (rf/sub [:wallet-connect/current-request-network])
+        {:keys [max-fees-fiat-formatted
+                error-state]} (rf/sub [:wallet-connect/current-request-transaction-information])]
```

Comment:
It looks a bit weird that we subscribe to the data inside the base modal. I'm guessing it would have to be in the parent component, since the sign-message modal doesn't need the transaction/network information at all.

Reply:
I'd also recommend making this component pure. Will help us when we expand to full-scope post 2.31.

I also tend to breakdown larger components into smaller ones (like you have done with `header`, there can also be an `alert`, `slider` and `txn-data`). Right now this is not a huge component, so it's fine. But there is a potential for complexity to grow. 

If the component is broken down, the next developer is more likely to follow the same pattern.

Reply:
Updated and fixed!

Reply:
It's not about purity though, it's about the fact that this component is simply not needed.